### PR TITLE
Migrate code to ES6 and gulp

### PR DESCRIPTION
**Pull request by Maxwell2022**

### CODE REVIEW

**Comment by Maxwell2022** [comment]
Not sure why hound is complaining, it's a valid JSON `.eslintrc` config. Also I received like 100 notification emails :/

**Comment by Maxwell2022** [comment]
Breaking change: Response is returning the body directly for successful requests or failed request.
Anything else from the response is not really needed.

**Comment by cdm-ium** [comment]
@Maxwell2022 i'll try to help you get this over the line so we can merge. First let's see if we can figure out the hound issue.

**Comment by cdm-ium** [comment]
@Maxwell2022  could you possibly push your branch so I can plan around with the build and see if I can get things to behave better.

File: dist/api/Accounts.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+Object.defineProperty(exports, "__esModule", {
+  value: true
+});
+
+var _BankAccounts = require('./BankAccounts');
```
Comment (cdm-ium) [inline dist/api/Accounts.js]: If we're going es6, any reason not to be using `const` here?

File: dist/api/Accounts.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  update(accountId, data) {
+    const headers = this.authenticator.secureHeaders('post', this.uri(accountId));
+    return this.request('post', { uri: this.uri(accountId), headers: headers, body: JSON.stringify(data) });
+  }
+
+  bankAccounts(accountId) {
+    return new _BankAccounts2.default(this.host, this.authenticator, this.uri(accountId));
+  }
+};
+exports.default = Accounts;
```
Comment (cdm-ium) [inline dist/api/Accounts.js]: Is there something I'm missing here, why do you need to bind `Accounts` with `let` above to assign it to defaults here.

Shouldn't 

```
class Accounts extends _Resource2.default {

}

exports.default = Accounts;
```
be fine?

File: dist/api/Resource.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
+'use strict';
```
Comment (cdm-ium) [inline dist/api/Resource.js]: At least a comment about the distinction between `Resource` and `Resource2` would be nice.

Comment (Maxwell2022) [inline dist/api/Resource.js]: Hey @cdm-ium ;) You commented on the `dist` folder which is the transpilled version. My code is in `src`

**Pull request closed without merging**